src/wp-includes/post.php

```diff
@@ -512,6 +520,7 @@ function create_initial_post_types() {
 				'filter_items_list'     => __( 'Filter Navigation Menu list' ),
 				'items_list_navigation' => __( 'Navigation Menus list navigation' ),
 				'items_list'            => __( 'Navigation Menus list' ),
+				'_edit_link'            => $navigation_post_edit_link, /* internal use only. don't use this when registering your own post type. */
```

Comment:
Actually, I just realised that this should be outside the  `labels` array.

Could we also add a test to [getEditPostLink](https://github.com/WordPress/wordpress-develop/blob/c40e0c6f816ed27ba8b75462632498a36f624181/tests/phpunit/tests/link/getEditPostLink.php)?

Something like this would work

```php
	/**
	 * Tests getting the edit post link for `wp_navigation` post type.
	 *
	 * @ticket 58589
	 */
	public function test_get_edit_post_link_for_wp_navigation_post_type() {
		$navigation_post = self::factory()->post->create_and_get(
			array(
				'post_type'    => 'wp_navigation',
				'post_name'    => 'wp_navigation',
				'post_title'   => 'My Navigation Post Type',
				'post_content' => 'Content',
				'post_excerpt' => 'Description of my navigation',
			)
		);

		$post_type_object = get_post_type_object( $navigation_post->post_type );
		$link_default     = admin_url( sprintf( $post_type_object->_edit_link, $navigation_post->ID ) );

		$this->assertSame( $link_default, get_edit_post_link( $navigation_post ) );
	}
```

You can run them via `npm run test:php -- --filter Tests_Link_GetEditPostLink`

Reply:
Updated positioning. Added tests

Reply:
Personally I'd prefer we didn't use WP functions to generate the "expected" in the assertion. It would be preferable to use hardcoded strings.

However, I've stuck with the status quote and existing patterns in the tests.